packages/core/src/dispatcher.ts

```diff
@@ -85,11 +107,20 @@ export default class Dispatcher extends EventEmitter {
   async register (document: Document): Promise<void> {
     this._documents[document.id.toString()] = document
     // request tip
-    const payload = { typ: MsgType.QUERY, id: document.id.toString(), doctype: document.doctype.doctype }
+    const payload = await this._buildQueryMessage(document)
     this._ipfs.pubsub.publish(this.topic, JSON.stringify(payload))
     this._log({ peer: this._peerId, event: 'published', topic: this.topic, message: payload })
   }
 
+  async _buildQueryMessage(document: Document): Promise<Record<string, any>> {
+    const message = { typ: MsgType.QUERY, doc: document.id.baseID.toString() }
+
+    // Add 'id' to message.  'id' is the multihash of the rest of the query object canonicalized
+    // using dag-cbor
+    const id: Uint8Array = dagCBOR.util.serialize(message)
```

Comment:
This is based on this line from CIP-71: `unique identifier of the query, multihash of the query object without the id property canonicalized using dag-cbor`
This was my guess as to what that meant, please let me know if this wasn't what you had in mind @oed